templates/client/html/catalog/detail/body.php

```diff
@@ -620,3 +620,30 @@
 	</div>
 
 <?php endif ?>
+
+<?php
+// preselection of a variant in a selection article
+if (isset($_GET['productId']) && is_numeric($_GET['productId'])):
+
+    foreach( $this->detailProductItem->getRefItems( 'product', null, 'default' ) as $prodid => $product ) :
+
+        if ($prodid == $_GET['productId']):
+
+            $attributes = $product->getRefItems('attribute',null,'variant');
+            $js = '';
+            foreach($attributes as $attribId => $attribute):
+                $js .= '"' .  $attribute->getType() . '": "' . $attribute->getId() . '",
+                ';
+            endforeach;
```

Comment:
Don't build a JSON-like string yourself but create an array of key/value pairs and use `json_encode()` afterwards